![](Database_Page_02.png)




![](titleproject.png)

![](nameproject.png)

# Excecutive Summary

<span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;">ในอดีตร้านขายยาคือหนึ่งในสถานที่ที่เราสามารถหายาเพื่อบรรเทาอาการเจ็บป่วยได้ โดยมีเภสัชกรเป็นคนให้คำแนะนำ แต่ในปัจจุบันกระแส E-Commerce มีการเติบโตอย่างรวดเร็วไม่เว้นแม้แต่ร้านขายยา อีกทั้งร้านขายยาไม่ได้ขายเพียงแค่ยาอีกต่อไป ยังมีการขายเครื่องสำอาง ผลิตภัณฑ์รักษาสุขภาพ และอื่นๆ อีกมากมาย ยกตัวอย่าง เช่น ร้าน Boost และ Watsons ที่เห็นได้ตามปกติทั่วไป&nbsp;</span>   

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Hello Pharmacy เป็นตัวอย่างที่ดีของธุรกิจร้านขายยาในการนำมาทำฐานข้อมูลเพื่อศึกษาและเก็บข้อมูลมาประยุกต์ใช้ในเชิงธุรกิจ เพื่อให้ทราบถึงการจัดการข้อมูลต่างๆ ไม่ว่าจะเป็นข้อมูล</span><span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;">สินค้า ข้อมูลลูกค้า และคำสั่งซื้อภายในร้าน ที่จะสามารถนำไปพัฒนาและปรับปรุงสินค้าและบริการให้ตอบโจทย์ลูกค้ามากที่สุด</span>

<span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;"><br></span>

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">ขอบเขตการดำเนินงาน</span>

- <span style="background-color: transparent; font-size: 11pt; white-space: pre-wrap;">ลูกค้ามีทั้งหมด 1,000 คน โดยทุกคนเป็นสมาชิกของร้าน Hello Pharmacy</span>
- <span style="background-color: transparent; font-size: 11pt; white-space: pre-wrap;">เก็บข้อมูลตั้งแต่ 1 มกราคม จนถึง 31 ธันวาคม ของปี 2020&nbsp;</span> 
- <span style="background-color: transparent; font-size: 11pt; white-space: pre-wrap;">ฐานข้อมูลนี้จัดการระบบสินค้าคงคลังแค่บางส่วนเท่านั้น</span>

# Background Information

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Hello Pharmacy เป็นร้านขายยาสมัยใหม่ที่มีช่องทางการขายทั้งในสาขา ซึ่งมีมากกว่า 20 สาขา ในรูปแบบ franchise ทั่วประเทศ และมีการจัดจำหน่ายในช่องทาง online ผ่านทาง website ซึ่งมีการจัดส่งสินค้าทั่วประเทศจากหลากหลายบริษัทชั้นนำ จึงสามารถจัดส่งสินค้าออกได้ทุกวัน ไม่เว้นวันหยุดนักขัตฤกษ์ โดยสินค้าที่จัดจำหน่ายเป็นสินค้าที่มีการคัดเลือกผลิตภัณฑ์ที่มีคุณภาพ สินค้าเวชภัณฑ์ผ่านการทดลองใช้จากฝ่าย R&amp;D ของทางร้านให้ได้มาตรฐาน และมีการตรวจสอบคุณภาพอยู่เป็นประจำ ซึ่งทางร้านมีสินค้าหลากหลายประเภท ทั้งยา ผลิตภัณฑ์เครื่องสำอาง ผลิตภัณฑ์รักษาสุขภาพ และผลิตภัณฑ์ประเภทอื่นๆ อีกมากมาย</span>

# Database Design

![](.\table1.jpg) ![](.\table2.jpg)

# ข้อ 1 : วิเคราะห์กลุ่มเป้าหมาย (Target Customer)

In [1]:
with cte as(
    SELECT age,
        case
            WHEN age BETWEEN 0 and 10 then ' 0 - 10'
            when age BETWEEN 11 and 20 then '11 - 20'
            when age BETWEEN 21 and 30 then '21 - 30'
            when age BETWEEN 31 and 40 then '31 - 40'
            when age BETWEEN 41 and 50 then '41 - 50'
            when age BETWEEN 51 and 60 then '51 - 60'
            when age BETWEEN 61 and 70 then '61 - 70'
            ELSE '- '
        end as "target_age"
        from customer
),
cte2 as(
    select target_age , COUNT("target_age") as quantity
    from cte
    GROUP BY target_age
 ),
 cte3 as (
      select * ,
    sum(quantity) over(ORDER BY target_age) cumulative_total, 
    sum(quantity) over() grand_total,
    count(quantity) over()
    from cte2
 )

SELECT 
    target_age, 
    quantity, 
    to_char(quantity*100 /grand_total ,'999D99"%"') as percent 
    from cte3

union

select 'total', sum(quantity), ' 100.00%'
    from cte3
    ORDER BY target_age


(7 row(s) affected)

Total execution time: 00:00:00.947

target_age quantity percent 11 - 20 59 5.90% 21 - 30 170 17.00% 31 - 40 204 20.40% 41 - 50 176 17.60% 51 - 60 184 18.40% 61 - 70 207 20.70% total 1000 100.00%

![](graph_target_project.png)
**แผนภาพเปรียบเทียบจำนวนลูกค้าในแต่ละช่วงอายุ**

## สรุปผลข้อมูลข้อ1

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">ตารางที่ 1 สร้างมาจากตาราง customer โดย คอลัมน์ “target_age” คือช่วงอายุของลูกค้า แบ่งตามช่วงวัย คอลัมน์ “quantity” คือจำนวนของลูกค้าที่อายุ อยู่ในช่วงดังกล่าว และ คอลัมน์ “percent” คือเปอร์เซ็นต์ของจำนวนลูกค้าในช่วงอายุนั้นๆ เทียบกับจำนวนลูกค้าทั้งหมด </span> <span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;">สามารถวิเคราะห์กลุ่มลูกค้าเป้าหมายได้ว่า</span>

- <span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">ลูกค้าใน<b>ช่วงอายุ 61-70 ปี</b> หรือก็คือลูกค้าในกลุ่มหลังเกษียณหรือกลุ่มสูงอายุ มีจำนวนมากถึง <b>20.70% </b>ของลูกค้าทั้งหมด</span>
- <span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">ลูกค้าใน<b>ช่วงอายุ&nbsp;</b></span> <span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;"><b>31-40 ปี</b> หรือก็คือลูกค้าในกลุ่มผู้ใหญ่ตอนต้น </span> <span style="color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 14.6667px; white-space: pre-wrap;">มีจำนวนมากถึง<b> 20.40%</b> </span> <span style="color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 14.6667px; white-space: pre-wrap;">ของลูกค้าทั้งหมด</span>
- <span style="color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 14.6667px; white-space: pre-wrap;">ลูกค้าใน</span><span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;"><b>ช่วงอายุ 51-60 ปี</b> หรือก็คือลูกค้าในกลุ่มผู้ใหญ่ตอนปลาย&nbsp; มีจำนวนมากถึง <b>18.40% </b>ของลูกค้าทั้งหมด</span>

<span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;">ดังนั้นจากข้อมูลที่วิเคราะห์มาข้างต้น สามารถสรุปได้ว่ากลุ่มลูกค้าของ Hello Pharmacy คือกลุ่มผู้ใหญ่ตอนต้นจนถึงกลุ่มผู้สูงอายุ เนื่องจากตั้งแต่วัยผู้ใหญ่ตอนต้นขึ้นไป เป็นวัยที่เริ่มเข้าสู่ช่วงของการดูแลตัวเองและรักษาสุขภาพมากขึ้น ทั้งในทางบำรุงร่างกาย เสริมความงาม หรือเพื่อผลเฉพาะทางในด้านโรคต่างๆ อีกทั้งเป็นวัยที่ปัจจัยด้านราคาไม่ค่อยมีผลต่อการเลือกซื้อสินค้า และถ้าเป็นเรื่องที่เกี่ยวกับสุขภาพ คนส่วนใหญ่มักจะใช้จ่ายไม่ว่าจะราคาเท่าไหร่ก็ตาม&nbsp;</span> 

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากข้อมูลข้างต้น <b style="">สามารถนำมาพัฒนาแผนการตลาดของ Hello Pharmacy</b> ได้ เช่น การทำการตลาดกับกลุ่มผู้ใหญ่ตอนต้นและตอนปลาย ด้วยการโฆษณาผ่าน social media โดยเฉพาะ Facebook ควบคู่ไปกับ การโฆษณาผ่านหน้าร้าน ใช้ทั้งช่องทาง online และ offline จัดทำโปรโมชั่น ให้เหมาะกับกลุ่มลูกค้าเหล่านี้เช่น สินค้าประเภทวิตามินและอาหารเสริม อุปกรณ์ซัพพอร์ต หรือผลิตภัณฑ์ดูแลผิวและความงามเป็นต้น</span>

# ข้อ 2 : วิเคราะห์สินค้าที่ขายดี/ไม่ดี อย่างละ 3 อันดับ เพื่อนำมาจัดการสินค้าคงคลัง

## 2.1 สินค้าขายดี

In [ ]:
with a as (
    select productid, name from product
),

b as (
    SELECT productid, sum(qty) qty from orderdtl
    GROUP BY productid
)

    SELECT a.productid, a.name, b.qty 
    from a inner join b on a.productid=b.productid
    ORDER BY qty  desc limit 3

## 2.2 สินค้าขายไม่ดี

In [ ]:
with a as (
    select productid, name from product
),

b as (
    SELECT productid, sum(qty) qty from orderdtl
    GROUP BY productid
)

    SELECT a.productid, a.name, b.qty 
    from a inner join b on a.productid=b.productid
    where b.qty <5
    ORDER BY qty  

## สรุปผลข้อมูลข้อ2

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;" id="docs-internal-guid-8d35189d-7fff-9ce8-0414-2589fe0f0bda">จากตารางที่ 2.1 และ 2.2 ที่สร้างมาจากตาราง orderdtl และตาราง product โดยคอลัมน์ “productid” คือรหัสสินค้า คอลัมน์ “name” คือชื่อสินค้า และคอลัมน์ “qty” คือปริมาณการสั่งซื้อสินค้าแต่ละชนิดในปี 2020</span>

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากผลลัพธ์ของตารางข้างต้น ทำให้ทราบถึงจำนวนสินค้าที่ขายได้มากที่สุดและน้อยที่สุด 3 อันดับแรกในปี 2020 ทำให้ Hello Pharmacy <b style="">สามารถนำมาวิเคราะห์ถึงการจัดการสินค้าคงคลังได้ </b>โดยดูว่าสินค้าชนิดไหนที่ขายดี ขายได้ปริมาณที่มาก ทางร้านอาจสั่งสินค้ากับทาง supplier มากขึ้น หรือถ้าหากสินค้าไหนที่ขายได้ในปริมาณที่น้อยอาจจะลดปริมาณในการสั่งลง การจัดการสินค้าคงคลังจะทำให้ธุรกิจมีสภาพคล่องมากขึ้น&nbsp; ง่ายต่อคาดคะเนเลือกพื้นที่เก็บสินค้าได้อย่างมีประสิทธิภาพมากขึ้น และสามารถตอบสนองความต้องการของลูกค้าได้ทันท่วงที นอกจากนั้นยังสามารถทำให้ธุรกิจได้ส่วนลดเพิ่มขึ้น จากการที่ซื้อในปริมาณมากต่อครั้ง</span>

# ข้อ 3 : ความนิยมในการสั่งซื้อสินค้าผ่านช่องทาง online ในแต่ละจังหวัด

In [ ]:
with cte as (
    select *, b.province “province”
    from orderhdr a inner join customer b using(cid)
),
cte1 as (
    select province, count(*) cnt
    from cte
    where channelid = 'ONLINE'
    group by province
    order by cnt desc
),
    cte2 as (
    Select row_number() over(order by cnt desc) "rank", * 
    from cte1
)
select * from cte2 limit 5

## สรุปผลข้อมูลข้อ3

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">ตารางที่ 3 สร้างมาจากตาราง orderhdr โดยคอลัมน์ “rank” คืออันดับของจังหวัดที่มีการสั่งซื้อสินค้าออนไลน์จากมากไปน้อย คอลัมน์ “province” คือจังหวัดต่างๆ ในประเทศไทย และคอลัมน์ “cnt” คือจำนวนคำสั่งซื้อสินค้าออนไลน์ในปี 2020</span>

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากตารางที่ 3 ทำให้ทราบถึง จังหวัดในประเทศไทยที่มีลูกค้าทำการสั่งซื้อสินค้าออนไลน์มากสุดเป็น 5 อันดับแรก จึงสามารถวิเคราะห์ถึงการพิจารณาเปิดสาขาหรือจุดกระจายสินค้าเพิ่ม ซึ่ง<i style=""><u style="">จังหวัดที่มีการสั่งซื้อสินค้าออนไลน์มากที่สุดคือ กรุงเทพมหานคร</u></i> โดยมียอดคำสั่งซื้อรวมทั้งหมด 348 คำสั่งซื้อ และร้าน Hello Pharmacy มีสาขาอยู่ในกรุงเทพมหานครเพียง 3 สาขา ด้วยข้อมูลที่กล่าวมานี้จึงทำให้<b style="">ทางร้านควรตัดสินใจพิจารณาในการเปิดสาขาเพิ่มเติมที่กรุงเทพมหานคร</b> เพื่อเป็นการตอบสนองความต้องการในการซื้อสินค้าของลูกค้าและเป็นการเพิ่มความสะดวกในการเลือกซื้อสินค้าหน้าร้านให้แก่ลูกค้าในกรุงเทพมหานครอีกด้วย</span>

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">นอกจากนี้<u><i>จังหวัดที่มีการสั่งซื้อสินค้าออนไลน์รองลงมาคือ จังหวัดชุมพร และเชียงใหม่</i></u> โดยมียอดคำสั่งซื้อรวมทั้งหมด 155 และ 131 คำสั่งซื้อตามลำดับ แต่เนื่องจากจังหวัดชุมพรนั้นอยู่ในภาคใต้ และภาคใต้มีร้าน Hello Pharmacy อยู่แล้ว 5 สาขา กับทางจังหวัดเชียงใหม่ที่มีสาขาของร้าน Hello Pharmacy อยู่แล้วเช่นกัน <b style="">จึงทำให้ควรไปพิจารณาที่จังหวัดที่มีการสั่งซื้อสินค้าออนไลน์รองลงมาเป็นลำดับที่ 4 คือจังหวัดบุรีรัมย์ </b>มียอดคำสั่งซื้อรวมทั้งหมด 108 คำสั่งซื้อ เนื่องจากมีร้าน Hello Pharmarcy อยู่เพียง 2 สาขาที่อยู่ในภาคกลางและไม่ได้อยู่ในกรุงเทพมหานคร การเปิดสาขาที่จังหวัดบุรีรัมย์เพิ่มเติมจึงมีความน่าสนใจ แต่ทั้งนี้ควรพิจารณายอดขายของปีอื่นๆ ประกอบเพิ่มเติมด้วย</span>

# ข้อ 4 : ประเภทสินค้าที่ขายได้ในแต่ละไตรมาส

In [ ]:
with a as(
    SELECT  * FROM orderdtl 
),
b as(
    SELECT orderid, date FROM orderhdr
),
c as(
    SELECT date_part('quarter', date)::int , * 
    FROM a join b using(orderid)
),
d as(
    SELECT productid, descrth FROM product
),
e as(
    SELECT date_part, descrth, sum(qty) 
    FROM c join d USING(productid) 
    GROUP BY date_part,descrth 
    ORDER BY date_part, sum(qty) desc 
),
t as(   
    SELECT 
        rank() over (PARTITION by date_part ORDER BY sum DESC) rank, *
        FROM e)

SELECT rank "Rank", date_part "Quarter", descrth "Type", sum "Quantity" FROM t where rank <=3


## สรุปผลข้อมูลข้อ4

<span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;">ตารางที่ 4 สร้างมากจากตาราง orderhdr, ตาราง orderdtl และตาราง product โดยคอลัมน์ “Rank” คืออันดับสินค้าขายดีจากมากไปน้อยของแต่ละไตรมาส คอลัมน์ “Quarter” คือไตรมาสที่เกิดคำสั่งซื้อสินค้า คอลัมน์ “Type” คือประเภทของสินค้า และคอลัมน์ “Quantity” คือปริมาณสินค้าที่ขายได้แยกตามเดือน</span>

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากตารางที่ 4 ทำให้ทราบได้ว่า ประเภทสินค้าที่ขายดีเป็นอันดับแรกและอันดับสองในทุกไตรมาส คือ <u><i>ประเภทผลิตภัณฑ์ดูแลผิว/ความงาม และประเภทคุณแม่และเด็ก</i></u> ตามลำดับ จึงสามารถคาดเดาได้ว่าสินค้าทั้ง 2 ประเภทนี้เป็นที่นิยมของคนไทยมาก จากข้อมูลข้างต้น ทางร้าน Hello Pharmacy <b>อาจเพิ่มการสั่งซื้อและผลักดันการขายสินค้าทั้ง 2 ประเภทนี้ </b>เพื่อให้สามารถเพิ่มยอดขายและรายได้ให้กับทางร้าน</span>

# ข้อ 5 : จำนวนคำสั่งซื้อแต่ละสาขา

In [ ]:
select branch.branchid "Branch ID", branch.descr as "Branch" , count(orderhdr.orderid) as "No_Order"
    from orderhdr inner join branch 
    on orderhdr.branchid = branch.branchid
    GROUP BY "Branch ID" , "Branch"
    ORDER BY "No_Order" desc

## สรุปผลข้อมูลข้อ5

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากตารางที่ 5 ซึ่งสร้างมาจากตาราง branch และตาราง orderhdr โดยคอลัมน์ “Branch ID” คือ ID ของแต่ละสาขาของร้านเรา คอลัมน์ “branch” คือ ชื่อของแต่ละสาขา หรือก็คือแหล่งที่ตั้งของแต่ละสาขา และคอลัมน์“no_order” คือจำนวนคำสั่งซื้อที่เกิดขึ้นในแต่ละสาขา</span>

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">โดยจากคำสั่งที่ใช้สืบค้นสำหรับแสดงผลการเรียงลำดับสาขาที่มีจำนวนคำสั่งซื้อที่มากที่สุดไปยังสาขาที่มีคำสั่งซื้อน้อยสุด สามารถวิเคราะห์ได้ว่า <i>สาขา ท็อปส์ พลาซ่า พะเยา เป็นสาขาที่มีจำการซื้อมากที่สุดจากทั้งหมด 20 สาขาของเรา และสาขาโลตัสระยองเป็นสาขาที่มีจำนวนการซื้อน้อยที่สุด</i></span>

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากข้อมูลข้างต้นสามารถ ทำให้ทาง<b>ฝ่ายจัดการร้านค้านำมาวางแผนเรื่องของจำนวนสินค้า</b>ได้เพื่อให้สามารถตอบสนองความต้องการของลูกค้าได้พอดี</span>

# ข้อ 6 : ช่องทางการชำระเงิน

In [ ]:
select payment.descr as "Payment" , count(orderhdr.orderid) as "qty"
    from payment inner join orderhdr 
    on orderhdr.paymentid = payment.paymentid
    GROUP BY  "Payment"
    ORDER BY "qty" desc

## สรุปผลข้อมูลข้อ6

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากตารางที่ 6 คอลัมน์ “Payment” คือ ช่องทาง การชำระเงินที่ลูกค้าใช้ และคอลัมน์ “Qty” คือจำนวนครั้งที่ลูกค้าใช้ช่องทางการชำระเงิน ซึ่งการชำระเงินของทางร้านที่มีบริการให้ลูกค้ามีอยู่ด้วยกัน 4 ช่องทาง ได้แก่ ชำระผ่านบัตรเครดิต Visa Card, บัตรเครดิต Master Card, ชำระผ่านการโอนเงิน (บริการ Promptpay) และชำระด้วยเงินสด&nbsp;</span> 

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากข้อมูลข้างต้นพบว่าการชำระเงินผ่านบัตรเครดิต <i>Visa Card มีจำนวนการใช้มากที่สุด</i> ทางร้านสามารถนำข้อมูลจำนวนการชำระเงินนี้<b>มาจัดทำโปรโมชั่นต่างๆ ร่วมกับ Visa Card ได้</b> เพื่อเป็นการดึงดูดลูกค้าอีกช่องทางหนึ่ง</span>

# ข้อ 7 : ลูกค้าที่เป็นผู้สนับสนุนหลักของร้าน

In [ ]:
WITH a as(
    SELECT cid,SUM(total) 
    FROM orderhdr
    GROUP BY cid
),
b as (
    SELECT cid,cfname,clname 
        FROM customer
)
SELECT cid "CustomerID",cfname "Fname",clname "Lname", to_char(sum, '9G999G999') "Total" FROM a join b 
USING(cid)
ORDER BY SUM desc LIMIT 200

## สรุปผลข้อมูลข้อ7

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">ตารางที่ 7 สร้างมาจากตาราง orderhdr และตาราง customer โดยตารางกล่าวถึงอันดับลูกค้าสมาชิกที่เป็นผู้สนับสนุนหลักของร้าน คอลัมน์ “CustomerID” คือรหัสลูกค้า คอลัมน์ “Fname” คือชื่อต้นของลูกค้า คอลัมน์ “Lname” คือนามสกุลของลูกค้า และคอลัมน์ “Total” คือยอดซื้อของลูกค้าทั้งหมด</span>

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากตารางที่ 7 นั้นเป็นการสร้างเพื่อหายอดใช้จ่ายของลูกค้าว่าลูกค้าคนใดที่เป็นสมาชิกกับทางร้านแล้ว มียอดใช้จ่ายสูงสุด 200 อันดับในปีนั้น จากสมาชิก 1,000 คน<b>จะได้รับ Gift Voucher เป็นของตอบแทน โดยใช้อิงหลักทฤษฎี 20:80</b> และเมื่อได้ทำการสืบค้นจากฐานข้อมูลก็พบว่า <i>ลูกค้าที่เป็นสมาชิกและมียอดค่าใช้จ่ายสูงสุดมีค่ามากถึง 19,557 บาท</i> จุดประสงค์ของการจัดทำตารางนี้ก็เพื่อเป็นการสร้าง <b>Brand Loyalty</b> แก่ลูกค้าที่เป็นสมาชิกกับทางร้าน ให้รู้สึกว่าทางร้านนั้นใส่ใจกับลูกค้าและกลับมาอุดหนุนสินค้าของร้านค้าอีก</span>

# ข้อ 8 : ยอดขายจากการซื้อหน้าร้านและช่องทาง online

In [ ]:
WITH a as(
SELECT date_part('month', date) ::INT "month",  sum(Total) "sqty"
    FROM orderhdr
    WHERE channelid = 'SHOP'
    GROUP BY channelid, MONTH
    ORDER BY MONTH
),
b as(
SELECT date_part('month', date) ::INT "month", sum(Total) "oqty"
    FROM orderhdr
    WHERE channelid = 'ONLINE'
    GROUP BY channelid, MONTH
    ORDER BY MONTH
),
 c as (
    SELECT a.month"month",a.sqty"shop",b.oqty"online" from a JOIN b using(month)

)

SELECT month, to_char(shop, '9G999G999') "Shop", to_char(Online, '9G999G999' ) "online" from c 
UNION
select null,to_char(sum(shop), '9G999G999'),to_char(sum(Online), '9G999G999') from c
ORDER BY month


## สรุปผลข้อมูลข้อ8

<span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap;">จากตารางที่&nbsp; 8&nbsp; ซึ่งสร้างมาจากตาราง orderhdr โดยตารางกล่าวถึงยอดขายจากการซื้อหน้าร้านและช่องทาง online คอลัมน์ “month” คือลำดับเดือน คอลัมน์ “Shop” คือรายได้รวมจากหน้าร้าน และคอลัมน์“lname” คือรายได้รวมจากช่องทาง online&nbsp;</span> 

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;"><span style="background-color: transparent; font-size: 11pt;">ตารางที่ 8 นั้นเป็นการสร้างเพื่อหายอดขายจากการซื้อหน้าร้านและช่องทาง online และเมื่อได้ทำการสืบค้นจากฐานข้อมูลก็พบว่า <i>ยอดขายจากการซื้อหน้าร้านนั้นมากกว่าช่องทาง online</i> ทำให้ทางร้าน Hello Pharmacy สามารถ<b>กำหนดงบ</b>ในการดูแลในแต่ละช่องทางและ<b>สังเกตุพฤติกรรมในการซื้อสินค้าของลูกค้า</b> เพื่อพัฒนาสินค้าและบริการให้ตรงกับความต้องการลูกค้ามากที่สุด</span>&nbsp;</span>

# ข้อ 9 : ยอดขายสินค้าแต่ละประเภท

In [4]:
with a as (
    select productid,sum(qty) qty from orderdtl
    GROUP BY productid
),

b as (
    select productid, typeid from product
),

c as (
    SELECT typid, descrth from type
)

SELECT c.typid, c.descrth, sum(a.qty) "quantity"
    from a inner join b on a.productid=b.productid inner join c on b.typeid=c.typid 
    GROUP BY typid,c.descrth
    ORDER BY quantity DESC 
    

(57 row(s) affected)

Total execution time: 00:00:00.012

typid descrth quantity beauty ผลิตภัณฑ์ดูแลผิว/ความงาม 1878 mnc คุณแม่และเด็ก 1624 house ยาสามัญประจำบ้าน 1262 alcohol ผลิตภัณฑ์ฆ่าเชื้อ แอลกอฮอลล์ 1245 excre การขับถ่าย 978 adult ผลิตภัณฑ์สำหรับผู้ใหญ่ 706 faidkit อุปกรณ์ปฐมพยาบาลและพลาสเตอร์ยา 698 pain ผลิตภัณฑ์แก้ปวดเมื่อย 670 bnj บำรุงกระดูกและข้อ 628 sunblock ครีมกันแดด/ลดฝ้า 624 hproduct สินค้าเพื่อสุขภาพ 593 workage ผู้ใหญ่และวัยทำงาน 570 vitamin วิตามินและอาหารเสริม 480 oralcare ผลิตภัณฑ์ดูแลช่องปาก 463 thermo ที่วัดไข้/วัดออกซิเจน 428 tear น้ำตาเทียม น้ำยาล้างตา 402 skincare บำรุงผิวพรรณ 337 mask หน้ากากอนามัย 321 mulcer ผลิตภัณฑ์แก้ร้อนใน 221 eye บำรุงสายตา 187 vijelly ขนมวิตามิน เยลลี่ 183 vibrain วิตามินเกี่ยวกับบำรุงสมอง 179 mwash น้ำยาบ้วนปาก 154 fungi ฆ่าเชื้อรา 151 sleep ช่วยในการนอนหลับ 151 bcream ครีมทาตัว 150 pregn ชุดตรวจครรภ์/วันไข่ตก 149 condom ถุงยางอนามัยและเจลหล่อลื่น 148 inhbalm ยาดมและยาหม่อง 143 pastille ลูกอม ชุ่มคอ 138 hair ผลิตภัณฑ์ดูแลเส้นผม 136 fclean ผลิตภัณฑ์ทำความสะอาดผิวหน้า 103 stnose บรรเทาอาการคัดจมูก 97 intiarea ผลิตภัณฑ์ดูแลจุดซ่อนเร้น 92 sksp ดูแลผิวหนังและสเปิร์ม 90 medsupp อาหารเสริมทางการแพทย์ 84 support อุปกรณ์ซัพพอร์ต 82 pcart ตลับยา/ที่บดยา 79 bclean ผลิตภัณฑ์ทำความสะอาดและดูแลผิวกาย 70 cfever บรรเทาปวด ลดไข้ เด็ก 69 serum เซรั่มและครีมบำรุง 68 hequip อุปกรณ์ตรวจวัดสุขภาพ 67 tradmed ยาแผนโบราณ 65 wart ผลิดภัณฑ์รักษาหูด 59 diges ท้องอืด ท้องเฟ้อ จุดเสียดแน่น อาหารไม่ย่อย 53 contact น้ำยาล้างคอนแทคเลนส์ 51 cough แก้ไอละลายเสมหะ 49 presgar เครื่องวัดความดัน/ระดับน้ำตาล 47 cardiac ผลิตภัณฑ์บำรุงหัวใจ 44 glove ถุงมือ 42 acne ผลิตภัณฑ์ดูแลสิว ฝ้า กระ 42 mosqui ผลิตภัณฑ์กันยุง 42 budd ชุดสังฆทาน 39 nsaline น้ำเกลือล้างจมูก 34 denture ครีมติดฟันปลอม 19 toothpa ยาสีฟัน 18 laxative ช่วยระบาย/ถ่ายไม่ออก 15

## สรุปผลข้อมูลข้อ9

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">ตารางที่ 9 สร้างมาจากตาราง orderdtl, ตาราง type และตาราง product โดยตารางที่ 9 กล่าวถึงยอดขายสินค้าแต่ละประเภท คอลัมน์ “typeid” คือชนิดหรือประเภทของสินค้า&nbsp; คอลัมน์ “descrth” คือการอธิบายเพิ่มเติมเกี่ยวกับสินค้านั้น&nbsp; คอลัมน์ “qty” คือยอดรวมจำนวนการซื้อ&nbsp;</span> 

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากตารางที่ 9 นั้นเป็นการสร้างเพื่อหาจำนวนยอดขายสินค้าแต่ละประเภท และเมื่อได้ทำการสืบค้นจากฐานข้อมูลพบว่า <i>ผลิตภัณฑ์ประเภทดูแลความงาม,คุณแม่และเด็ก และยาสามัญประจำบ้าน มีจำนวนการซื้อจำนวนมากที่สุดตามลำดับ</i> โดยจุดประสงค์ของการจัดทำตารางนี้ก็เพื่อเป็นการจัดคำสั่งซื้อ อัปเดตสต็อกและปรับปรุงสต็อกให้เพียงพอต่อความต้องการของลูกค้า ดังนั้นทาง</span> <span style="color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 14.6667px; white-space: pre-wrap;">Hello Pharmacy อาจทำการวิเคราะห์ต่อว่า สินค้าที่ขายดี 3 อันดับนี้ในการสั่งสินค้าครั้งถัดไป<b>ควรสั่งในปริมาณที่เพิ่มขึ้นหรือไม่</b> เพื่อที่จะได้รองรับความต้องการของลูกค้าในปีถัดไป หรือถ้าหากสินค้าประเภทไหนที่ขายไม่ดีภายในปีนี้ <b>ควรจะเอาออกมาทำโปรโมชั่นก่อนที่ยาจะหมดอายุ</b>หรือไม่ เพื่อไม่ให้มีสินค้าตกค้างในคลัง</span>

# ข้อ 10 : การจัดโปรโมชั่น

In [ ]:
select productid, name , status, promotion from product
    where status = 'out' and promotion = true

## สรุปผลข้อมูลข้อ10

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">ตารางที่ 10 ที่สร้างมาจาก ตาราง product โดยคอลัมน์ “productid” คือรหัสสินค้า คอลัมน์ “name” คือชื่อสินค้า คอลัมน์ “status” คือสถานะของสินค้าในคลัง ถ้า status มีค่าเท่ากับ in แปลว่า สินค้ายังคงอยู่ในคลัง แต่ถ้าหาก status มีค่าเท่ากับ out แปลว่า สินค้าไม่มีอยู่ในคลังื(หมด stock) และคอมลัมน์ “promotion” คือค่าที่บอกว่าสินค้าชิ้นนั้นเข้าร่วมโปรโมชั่นอยู่หรือไม่&nbsp;</span> 

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากผลลัพธ์ของตารางข้างต้นทำให้สรุปได้ว่า ณ ตอนนี้<i>มีสินค้าอยู่ 3 ชนิดที่เข้าร่วมโปรโมชั่นและขายจนหมด stock โดยไม่มีสินค้าเหลืออยู่ในคลัง</i> ทำให้ Hello Pharmacy สามารถนำมาวิเคราะห์ต่อได้ว่า การทำโปรโมชั่นกับสินค้าเหล่านี้ได้รับการตอบรับจากลูกค้าเป็นอย่างดีจนไม่เพียงพอต่อความต้องการของลูกค้า ทาง<b>บริษัทควรจะหาสินค้ามาเพิ่ม</b> หรืออาจจะ<b>ติดต่อกับ supplier</b> ว่าในการสั่งสินค้าครั้งถัดไป จะ<b>สั่งสินค้าใน lot size ที่ใหญ่ขึ้น ในราคาที่ถูกลง</b> ทางบริษัทจะได้มีต้นทุนที่ไม่สูงมาก และมีกำไรที่เพิ่มขึ้นจากการทำ promotion ต่อไป</span>

# Conclusion

<span style="font-size: 11pt; font-family: Sarabun, sans-serif; color: rgb(0, 0, 0); background-color: transparent; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">จากการสืบค้นและวิเคราะห์ข้อมูลทั้งหมดของร้าน Hello Pharmacy พบว่ากลุ่มเป้าหมายของทางร้าน ได้แก่ ลูกค้าวัยผู้ใหญ่ตอนต้นจนถึงผู้สูงอายุ โดยยอดขายส่วนใหญ่มาจากกรุงเทพมหานคร ทั้งจากการขายหน้าร้านและช่องทาง online ซึ่งสินค้าที่ได้รับความนิยมมาก คือสินค้าประเภทผลิตภัณฑ์ดูแลความงามและประเภทคุณแม่และเด็กตามลำดับ&nbsp;</span> 

<span style="background-color: transparent; color: rgb(0, 0, 0); font-family: Sarabun, sans-serif; font-size: 11pt; white-space: pre-wrap; text-indent: 36pt;">จากข้อมูลข้างต้น สามารถสรุปได้ว่า ทางร้านควรจัดทำโปรโมชั่นการขายสินค้าและการชำระงินให้ตรงกับความต้องการของลูกค้า และจัดทำแผนการตลาดเพื่อสนับสนุนการขายสินค้า ทั้งสินค้าที่ได้รับความนิยมเป็นอย่างมากอยู่แล้ว และสินค้าที่ต้องการเพิ่มยอดขายให้มากขึ้น รวมไปถึงมีการจัดการสินค้าคงคลังที่เหมาะสมกับการขายสินค้าที่เกิดขึ้น และควรมีแผนในการขยายสาขา เพื่อให้รองรับการเติบโตของตลาดและสังคมผู้สูงอายุในอนาคต</span>

# Reference lists

![](บรรณานุกรมใหม่.png)

# ERD
![](ERDร้านขายยา.jpg)